libgalois/src/FileGraph.cpp

```diff
@@ -708,5 +708,39 @@ FileGraph::iterator FileGraph::end() const {
   return iterator(nodeOffset + numNodes);
 }
 
+void FileGraph::initNodeDegrees() {
+  if (!this->node_degrees.size()) {
+    // allocate memory
+    this->node_degrees.create(this->numNodes);
+    // loop over all nodes, calculate degrees
+    galois::do_all(
+        galois::iterate(0ul, this->numNodes),
+        [&](unsigned long n) {
+          // calculate and save degrees
+          if (n != 0) {
+            this->node_degrees.set(n, this->outIdx[n] - this->outIdx[n - 1]);
+          } else {
+            this->node_degrees.set(n, this->outIdx[0]);
+          }
+        },
+        galois::loopname("FileGraphInitNodeDegrees"), galois::no_stats());
+  }
+}
+
+uint64_t FileGraph::getDegree(uint32_t node_id) const {
+  if (this->node_degrees.size()) {
```

Comment:
Add an assertion for this and change the documentation. This can be called within a parallel function. We should not be calculating it on-the-fly.

Reply:
in other words, make it getDegree fails if init isn't called beforehand?

Reply:
Yes.